src/h3/_cy/h3lib.pxd

```diff
@@ -48,9 +48,9 @@ cdef extern from "h3api.h":
         LinkedGeoLoop *_data_last "last"  # not needed in Cython bindings
         LinkedGeoPolygon *next
 
-    H3Index geoToH3(const GeoCoord *g, int res)
+    H3Index geoToH3(const GeoCoord *g, int res) nogil
```

Comment:
How do you determine which functions get `nogil`? Is it just the ones you want to optimize right now?

Reply:
I just added it to the functions I needed it for in this PR. We could probably add `nogil` to each of the functions in the API, as the `nogil` tag just promises that the function won't manipulate any Python objects, which is true for everything in the `h3-c` API.

I'll probably do a more thorough job of this when I tackle https://github.com/uber/h3-py/issues/140